File: src/014/p014.py
#!/usr/bin/env python


def collatz(n):
    seq = [n]
    while n > 1:
        if n % 2 == 0:
            n = n / 2
        else:
            n = 3 * n + 1
        seq.append(int(n))
    return seq


sizes = [len(collatz(i)) for i in range(1, int(1e6))]

print(sizes.index(max(sizes)) + 1)

File: src/014/p014.rb
def collatz(num)
  seq = [num]
  while num > 1
    num = num.even? ? num / 2 : 3 * num + 1
    seq.append(num.to_i)
  end
  seq
end

sizes = (1..1e6).map { |i| collatz(i).size }

puts sizes.index(sizes.max) + 1
